tests/lib/ringbuffer/src/main.c

```diff
@@ -443,6 +476,58 @@ void test_reset(void)
 	zassert_true(granted == RINGBUFFER_SIZE - 1, NULL);
 }
 
+#ifdef CONFIG_64BIT
+static uint64_t ringbuf_stored[RINGBUFFER_SIZE];
```

Comment:
Is this correct? The ring_buf_init doxygen says:

```
 * @param data Ring buffer data area (uint32_t data[size] or uint8_t data[size] for
 *	       bytes mode).
```

Reply:
Good catch.  Yeah, this is wasteful (though not incorrect as I read the code).  If you're going to use extra memory in a test like this, at least do it to put a guard area at the edges to check for overrruns. :)

Reply:
Hi @mbolivar-nordic @andyross , thanks for your comments, yeah, I totally agree that it's not allowed to use the array defined by uint64_t in normal. Notably, this is a test that the pointer size is equal to the ring buffer size. And the platforms are 32bits and 64bits, the pointer also. To keep with the same as pointer size, I use the #ifdef#endif to ensure that testcase can be implement. 

By the way, the ring buffer is implemented as a 'ring', so why do I need to put a guard area? Thanks again. :)

Reply:
Hi @mbolivar-nordic @andyross , thanks for your comments, yeah, I totally agree that it's not allowed to use the array defined by uint64_t in normal. Notably, this is a test that the pointer size is equal to the ring buffer size. And there are 32bits and 64bits about platforms(and pointer). To keep with the same as pointer size, I used the #ifdef#endif to ensure that the testcase is compatible with all platforms. 

And... It's actually wasteful, I find that the extra memory can't be used by ring buffer APIs in 64bits platform. And the test is just to testify that ring buffer with (32bits or 64bits)buffer can work. :)

Reply:
You don't need a guard area.  I was saying that **if** you were going to waste memory like this with an oversized buffer, one **good** reason (in a test like this) would be to use it to test for overruns by the code.

Reply:
Got it, thanks very much. :)